### ISSUE TITLE
Evaluate Algolia Search

### ISSUE THREAD

**Issue by bhaskarmelkani** [zalando-incubator/hexo-theme-doc#150]
# Evaluate [Algolia](https://www.algolia.com/ref/docsearch?utm_source=docsearch&utm_medium=link&utm_term=footer&utm_campaign=hexo.io) Search

Currently, the theme uses [lunr](https://lunrjs.com/) for searching inside the documentation. There are some pitfalls for using lunr [the generated index file(lunar.json) can grow super big with the data in the documentation, support for multiple languages like Chinese, etc].

This ticket is to discuss the pros and cons of this implementation in the theme and decide if its worth while to do this change and replace lunr with Algolia.

**Comment by miaojf** [comment]
https://community.algolia.com/docsearch/

**Comment by bhaskarmelkani** [comment]
[hexo.io](https://hexo.io/) also uses Algolia for in-site search.

Heres the issue for it. https://github.com/hexojs/site/issues/129

**Comment by rbarilani** [comment]
@miaojf @bhaskarmelkani Seems cool but if I understood correctly, algolia is a service as Google Search that requires to have your site crawled, which is fine for public sites but not for private sites.

So I say that we should support both (lunr.json) and algolia, up to the user to decide which one to use.

WDYT?

**Comment by bhaskarmelkani** [comment]
Yes, Algolia will provide both searches inside the document using the already generated index and also providing other functionalities like crawling.
There are some plugins to enable Algolia for hexo.
* https://github.com/oncletom/hexo-algolia
* https://github.com/LouisBarranqueiro/hexo-algoliasearch
